habitat/sims/pyrobot/pyrobot.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any
+from gym import Space, spaces
+
+from habitat.core.registry import registry
+from habitat.core.simulator import (
+    Config,
+    DepthSensor,
+    RGBSensor,
+    SensorSuite,
+    Simulator,
+)
+import pyrobot
+import numpy as np
+
+# TODO(akadian): remove the below pyrobot hack
+import sys
+
+ros_path = "/opt/ros/kinetic/lib/python2.7/dist-packages"
+if ros_path in sys.path:
+    sys.path.remove(ros_path)
+    import cv2
+sys.path.append(ros_path)
+
+
+@registry.register_sensor
+class PyRobotRGBSensor(RGBSensor):
+    def __init__(self, config):
+        super().__init__(config=config)
+
+    def _get_observation_space(self, *args: Any, **kwargs: Any):
+        return spaces.Box(
+            low=0,
+            high=255,
+            shape=(self.config.HEIGHT, self.config.WIDTH, 3),
+            dtype=np.uint8,
+        )
+
+    def get_observation(self, robot_obs):
+        obs = robot_obs.get(self.uuid, None)
+
+        assert obs is not None, "Invalid observation for {} sensor".format(
+            self.uuid
+        )
+
+        if obs.shape != self.observation_space.shape:
+            obs = cv2.resize(
+                obs,
+                (
+                    self.observation_space.shape[1],
+                    self.observation_space.shape[0],
+                ),
+            )
+
+        return obs
+
+
+@registry.register_sensor
+class PyRobotDepthSensor(DepthSensor):
+    def __init__(self, config):
+        if config.NORMALIZE_DEPTH:
+            self.min_depth_value = 0
+            self.max_depth_value = 1
+        else:
+            self.min_depth_value = config.MIN_DEPTH
+            self.max_depth_value = config.MAX_DEPTH
+
+        super().__init__(config=config)
+
+    def _get_observation_space(self, *args: Any, **kwargs: Any):
+        return spaces.Box(
+            low=self.min_depth_value,
+            high=self.max_depth_value,
+            shape=(self.config.HEIGHT, self.config.WIDTH, 1),
+            dtype=np.float32,
+        )
+
+    def get_observation(self, robot_obs):
+        obs = robot_obs.get(self.uuid, None)
+
+        assert obs is not None, "Invalid observation for {} sensor".format(
+            self.uuid
+        )
+
+        if obs.shape != self.observation_space.shape:
+            obs = cv2.resize(
+                obs,
+                (
+                    self.observation_space.shape[1],
+                    self.observation_space.shape[0],
+                ),
+            )
+
+        obs = np.clip(obs, self.config.MIN_DEPTH, self.config.MAX_DEPTH)
```

Comment:
Looks like it should be in `else` statement for `if self.config.MAX_DEPTH:`, otherwise normalization doesn't make sense as `self.config.MAX_DEPTH = 1` if `self.config.MAX_DEPTH == True`.

Reply:
I think the current implementation is correct, I will try to explain the usage below, please let me know if you think the interpretation is awry.

`MAX_DEPTH` and `NORMALIZE_DEPTH` are orthogonal. `MAX_DEPTH` specifies at what max depth rating should the depth sensor's observation be clipped. `NORMALIZE_DEPTH` flag controls whether the observations coming from the depth sensor should be in `[0, 1]` or `[MIN_DEPTH, MAX_DEPTH]`.